spec/classes/puppet_agent_install_spec.rb

```diff
@@ -7,28 +7,41 @@
       'include ::puppet'
     end
 
+
+    if Puppet.version < '4.0'
+      client_package = 'puppet'
+      additional_facts = {}
+    else
+      client_package = 'puppet-agent'
+      additional_facts = {:rubysitedir => '/opt/puppetlabs/puppet/lib/ruby/site_ruby/2.1.0'}
+    end
+
     context "on a RedHat family OS" do
-      let :facts do {
-        :osfamily => 'RedHat',
-        :concat_basedir => '/foo/bar',
-        :operatingsystemrelease => '6.6',
-        :puppetversion => Puppet.version,
-      } end
+      let :facts do
```

Comment:
It may be better to use ```on_supported_os['redhat-6-x86_64']``` so you also get the ```rubysitedir``` fact, which is now needed and fails tests.